src/main/java/org/folio/innreach/domain/entity/InnReachTransaction.java

```diff
@@ -84,6 +90,12 @@
     "JOIN FETCH h.pickupLocation " +
     "WHERE h.folioItemId = :folioItemId AND h.folioPatronId = :folioPatronId AND irt.state NOT IN (4, 11, 12, 13)";
 
+  public static final String FETCH_ASSOCIATED_QUERY_NAME = "InnReachTransaction.fetchAssociatedLoan";
+  public static final String FETCH_ASSOCIATED_QUERY = "SELECT irt FROM InnReachTransaction AS irt " +
+    "JOIN FETCH irt.hold AS h " +
+    "JOIN FETCH h.pickupLocation " +
+    "WHERE h.folioLoanId = :folioLoanId";
```

Comment:
This query should probably fetch only active transactions

Reply:
AND irt.state NOT IN (4, 11, 12, 13)